i2c_manager/i2c_manager.c

```diff
@@ -0,0 +1,321 @@
+/*
+
+SPDX-License-Identifier: MIT
+
+MIT License
+
+Copyright (c) 2021 Rop Gonggrijp. Based on esp_i2c_helper by Mika Tuupola.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <esp_log.h>
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/semphr.h"
+#include "freertos/task.h"
+#include <driver/i2c.h>
+
+#include "sdkconfig.h"
+
+#include "i2c_manager.h"
+
+static const char* TAG = I2C_TAG;
+
+static SemaphoreHandle_t I2C_FN(_local_mutex)[2] = { NULL, NULL };
+static SemaphoreHandle_t* I2C_FN(_mutex) = &I2C_FN(_local_mutex)[0];
+
+static const uint8_t ACK_CHECK_EN = 1;
+
+#if defined (CONFIG_I2C_MANAGER_0_ENABLED)
+	#if defined (CONFIG_I2C_MANAGER_0_PULLUPS)
+		#define I2C_MANAGER_0_PULLUPS 	true
+	#else
+		#define I2C_MANAGER_0_PULLUPS 	false
+	#endif
+
+	#define I2C_MANAGER_0_TIMEOUT 		CONFIG_I2C_MANAGER_0_TIMEOUT / portTICK_RATE_MS
+	#define I2C_MANAGER_0_LOCK_TIMEOUT	CONFIG_I2C_MANAGER_0_LOCK_TIMEOUT / portTICK_RATE_MS
+#endif
+
+
+#if defined (CONFIG_I2C_MANAGER_1_ENABLED)
+	#if defined (CONFIG_I2C_MANAGER_1_PULLUPS)
+		#define I2C_MANAGER_1_PULLUPS 	true
+	#else
+		#define I2C_MANAGER_1_PULLUPS 	false
+	#endif
+
+	#define I2C_MANAGER_1_TIMEOUT 		CONFIG_I2C_MANAGER_1_TIMEOUT / portTICK_RATE_MS
+	#define I2C_MANAGER_1_LOCK_TIMEOUT	CONFIG_I2C_MANAGER_1_LOCK_TIMEOUT / portTICK_RATE_MS
+#endif
+
+esp_err_t I2C_FN(_init)(i2c_port_t port) {
+
+	esp_err_t ret = ESP_OK;
+
+	if (I2C_FN(_mutex)[port] == 0) {
+
+		ESP_LOGI(TAG, "Starting I2C master at port %d.", (int)port);
+
+		I2C_FN(_mutex)[port] = xSemaphoreCreateMutex();
+
+		i2c_config_t conf;
```

Comment:
this variable needs to be zero initialized `i2c_config_t conf = {0}`.

Reply:
fixed